javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -674,11 +674,11 @@ public static Double toNumeric(Object o) {
             val = (Double)o;
         } else if (o instanceof String) {
             String s = ((String)o).trim();
-            double d;

```

Comment:
The change in this commit seemed like the cleanest way to make sure we don't attempt a date conversion on a string we already know has invalid characters. Otherwise both the `checkForInvalidNumericOrDatestringCharacters()` method and the `Double.parseDouble()` call were both potentially throwing a `NumberFormatException`, with no way to distinguish between them.